File: 03_patterns/swift/mvvm_clean_arc/Clean/DI/Dependencies.swift
import Foundation

enum Dependencies {
    static func makeTaskListViewModel() -> CleanTaskListViewModel {
        let localDataSource = TaskLocalDataSource()
        let repository = TaskRepositoryImpl(localDataSource: localDataSource)

        return CleanTaskListViewModel(
            getTasksUseCase: GetTasksUseCase(repository: repository),
            createTaskUseCase: CreateTaskUseCase(repository: repository),
            deleteTaskUseCase: DeleteTaskUseCase(repository: repository)
        )
    }
}

File: 03_patterns/flutter/lib/clean/di/dependencies.dart
import '../data/data_source/task_local_data_source.dart';
import '../data/repository/task_repository_impl.dart';
import '../domain/use_cases/create_task_use_case.dart';
import '../domain/use_cases/delete_task_use_case.dart';
import '../domain/use_cases/get_tasks_use_case.dart';
import '../presentation/view_models/clean_task_list_view_model.dart';

class Dependencies {
  static CleanTaskListViewModel makeTaskListViewModel() {
    final localDataSource = TaskLocalDataSource();
    final repository = TaskRepositoryImpl(localDataSource: localDataSource);

    return CleanTaskListViewModel(
      getTasksUseCase: GetTasksUseCase(repository: repository),
      createTaskUseCase: CreateTaskUseCase(repository: repository),
      deleteTaskUseCase: DeleteTaskUseCase(repository: repository),
    );
  }
}
